### PR TITLE
MULE-19167: Adding javadoc about why SchedulingStrategy is not an UnionType

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java [hunk 1]
```diff
@@ -125,10 +125,23 @@ private ParameterDeclaration declareSchedulingStrategyParameter(ClassTypeLoader
...
       ClassTypeLoader typeLoader = ExtensionsTypeLoaderFactory.getDefault().createTypeLoader();
-      extensionDeclarer.withImportedType(new ImportedTypeModel((ObjectType) typeLoader.load(SchedulingStrategy.class)));
+      extensionDeclarer.withImportedType(new ImportedTypeModel((ObjectType) loadSchedulingStrategyType(typeLoader)));
     }
   }
 
+  /**
+   * This method is just loading the SchedulingStrategy type from the core extension model, so it doesn't need to declare the
+   * subtypes (cron and fixed-frequency).
+   *
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java:135]: `<p>`

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java [hunk 2]
```diff
@@ -125,10 +125,23 @@ private ParameterDeclaration declareSchedulingStrategyParameter(ClassTypeLoader
 
     if (thereArePollingSources.get() && !isSchedulerAlreadyImported(extensionDeclarer.getDeclaration())) {
       ClassTypeLoader typeLoader = ExtensionsTypeLoaderFactory.getDefault().createTypeLoader();
-      extensionDeclarer.withImportedType(new ImportedTypeModel((ObjectType) typeLoader.load(SchedulingStrategy.class)));
+      extensionDeclarer.withImportedType(new ImportedTypeModel((ObjectType) loadSchedulingStrategyType(typeLoader)));
     }
   }
 
+  /**
+   * This method is just loading the SchedulingStrategy type from the core extension model, so it doesn't need to declare the
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java:133]: `@link`

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java [hunk 3]
```diff
@@ -125,10 +125,23 @@ private ParameterDeclaration declareSchedulingStrategyParameter(ClassTypeLoader
...
+   * subtypes (cron and fixed-frequency).
+   *
+   * It isn't implemented as an UnionType because of backwards compatibility (see MULE-19167 and MuleExtensionModelDeclarer).
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java:136]: `@link`, `@link`

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java [hunk 4]
```diff
@@ -125,10 +125,23 @@ private ParameterDeclaration declareSchedulingStrategyParameter(ClassTypeLoader
...
+   *
+   * @param loader The type loader.
+   * @return The scheduling-strategy parameter type.
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java:139]: `@code`

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java [hunk 5]
```diff
@@ -125,10 +125,23 @@ private ParameterDeclaration declareSchedulingStrategyParameter(ClassTypeLoader
...
     if (thereArePollingSources.get() && !isSchedulerAlreadyImported(extensionDeclarer.getDeclaration())) {
       ClassTypeLoader typeLoader = ExtensionsTypeLoaderFactory.getDefault().createTypeLoader();
-      extensionDeclarer.withImportedType(new ImportedTypeModel((ObjectType) typeLoader.load(SchedulingStrategy.class)));
+      extensionDeclarer.withImportedType(new ImportedTypeModel((ObjectType) loadSchedulingStrategyType(typeLoader)));
     }
   }
 
+  /**
+   * This method is just loading the SchedulingStrategy type from the core extension model, so it doesn't need to declare the
+   * subtypes (cron and fixed-frequency).
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java:134]: `@code`

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java (hunk 4 above)
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java:139]: `@code`

File: runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java [hunk 6]
```diff
@@ -241,17 +241,48 @@ private void declareScheduler(ExtensionDeclarer extensionDeclarer, ClassTypeLoad
...
 
     scheduler.onDefaultParameterGroup()
         .withOptionalParameter("disallowConcurrentExecution")
         .ofType(typeLoader.load(Boolean.class))
         .defaultingTo(false)
         .withExpressionSupport(NOT_SUPPORTED);
+  }
+
+  /**
+   * The scheduling-strategy type may be cron or fixed-frequency, and this hierarchy is implemented by using subtypes. It makes
```
Comment (elrodro83) [inline runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java:257]: `@code`

File: runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java [hunk 7]
```diff
@@ -241,17 +241,48 @@ private void declareScheduler(ExtensionDeclarer extensionDeclarer, ClassTypeLoad
...
+   *   &#64;code
+   *   <scheduler>
+   *     <scheduling-strategy>
+   *       <fixed-frequency params... />
+   *     </scheduling-strategy>
+   *   </scheduler>
+   * }
+   * </pre>
+   *
+   * If it were implemented as an UnionType (see MULE-19167) it would look like this, and it would break backwards:
```
Comment (elrodro83) [inline runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java:271]: `@link`

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by ndinu — APPROVED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**